File: 프로그래머스/1/12901. 2016년/2016년.py
def solution(a, b):
    answer = ''
    last_date = [31,29,31,30,31,30,31,31,30,31,30,31]
    date = ["SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"]
    
    period = 0
    for i in range(a-1):
        period += last_date[i]
        
    period += b - 1
    remain = period % 7
    
    return date[(5 + remain) % 7]



# 1월 1일 금
# 1월 4일 월
# 1월 8일 금
# 1월 6일 수 (6-1) % 7 = 5 => Fri + 3

# 2월 1일 : 1월 1일 = 1 2월 1일 = (31 + 1 - 1)일 째, 31 % 7 = 3, fri + 3 = "MON"

File: 프로그래머스/1/12901. 2016년/2016년.java
class Solution {
    public String solution(int a, int b) {
        String answer = "";
        int[] days = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31, 30, 31};
        String[] week = {"SUN","MON","TUE","WED","THU","FRI","SAT"};
        int index = 5;
        
        // a월 b일과 1월 1일 날짜 차이
        int p = 1;
        int q = b;
        for (int i = 0; i < a-1; i++) {
            q += days[i];
        }
        int k = q - p;
                
        int remain = k % 7;
        answer = week[(index + remain) % 7];
        
        
        return answer;
    }
}